torch/csrc/jit/ir/constants.cpp

```diff
@@ -108,6 +109,11 @@ c10::optional<Value*> tryInsertConstant(
     ss << val.toDevice();
     n->s_(attr::value, ss.str());
     n->output()->setType(DeviceObjType::get());
+  } else if (val.isGenerator()) {
+    auto generator = val.toGenerator();
+    n->ival_(attr::value, generator);
+    n->i_(torch::jit::Symbol::attr("seed"), generator.current_seed());
```

Comment:
are these the semantics we want, i.e. should we be resetting the seed? Will that produce the same behavior in eager as in torchscript?

i.e., this seems to suggest that on each iteration the generator will be producing the same results; whereas in eager mode I'm not sure whether we would always expect the generator to be reset on each iteration.

I guess part of the problem is that I'm not quite sure where we'd want to `tryInsertConstant` for a generator - do you have an example of where this is getting called?

Reply:
> are these the semantics we want, i.e. should we be resetting the seed? Will that produce the same behavior in eager as in torchscript?

This is a good point. And in fact, this seed setting logic is not needed if we are doing `torch.jit.script`. I added it because I noticed that `torch.jit.trace` doesn't reset the generator seed every time it is run.

Is it an acceptable limitation for now to say that one _has_ to use `torch.jit.script` if you want to use a generator in TorchScript?